### PR TITLE
Clear managed type map with dirties context

### PR DESCRIPTION
**Pull request by ng-galien**
When testing Spring app with @DirtiesContext we must refresh the mapping for managed types.

### CODE REVIEW

File: rsql-common/src/main/java/io/github/perplexhub/rsql/RSQLCommonSupport.java [hunk 1]
```diff
@@ -37,6 +37,7 @@ public RSQLCommonSupport() {
 	public RSQLCommonSupport(Map<String, EntityManager> entityManagerMap) {
 		if (entityManagerMap != null) {
 			RSQLCommonSupport.entityManagerMap.putAll(entityManagerMap);
+			RSQLCommonSupport.managedTypeMap.clear();
```
Comment (perplexhub) [inline rsql-common/src/main/java/io/github/perplexhub/rsql/RSQLCommonSupport.java:40]: Should we clear managedTypeMap in constructor?

Comment (ng-galien) [reply rsql-common/src/main/java/io/github/perplexhub/rsql/RSQLCommonSupport.java:40]: I found the problem in my company project that _sometimes_ must clear Spring context in IT test. This fix address this case but I'm not sure this covers all the possible side effects of the static cache.

Comment (perplexhub) [reply rsql-common/src/main/java/io/github/perplexhub/rsql/RSQLCommonSupport.java:40]: I agree the managedTypeMap must be cleared in some test. I think the cache should be cleared by @BeforeEach/@BeforeAll between the tests.
In some production code, managedTypeMap would be assigned before calling the constructor, i.e. the change would clear the predefined mapping.

Comment (ng-galien) [reply rsql-common/src/main/java/io/github/perplexhub/rsql/RSQLCommonSupport.java:40]: Yes I do it manually on affected existing projects but it was not trivial to find the problem.

Comment (perplexhub) [reply rsql-common/src/main/java/io/github/perplexhub/rsql/RSQLCommonSupport.java:40]:
```java
RSQLVisitorBase.setEntityManagerMap(getEntityManagerMap());
RSQLVisitorBase.setManagedTypeMap(getManagedTypeMap());
RSQLVisitorBase.setPropertyRemapping(getPropertyRemapping());
RSQLVisitorBase.setGlobalPropertyWhitelist(getPropertyWhitelist());
RSQLVisitorBase.setGlobalPropertyBlacklist(getPropertyBlacklist());
RSQLVisitorBase.setDefaultConversionService(getConversionService());	
```
We could implement above ```setXXX``` methods to reset and copy the caches.

**Comment by perplexhub** [comment]
How about implement in this way? https://github.com/perplexhub/rsql-jpa-specification/pull/128

**Comment by ng-galien** [comment]
Yes probably cleaner.

**Pull request closed without merging**